### PR TITLE
Implement core algorithm of target prediction

### PR DESCRIPTION
**Pull request by YunwooLee94**
I made a PR to  merge the old branch (utils_arrange) with "master" and create a new branch.
For the discussion of interfaces, I will make a new issue.

### CODE REVIEW

**Comment by icsl-Jeon** [comment]
Did you rebase the current master? If then, good to merge after modifying cmakelists little bit

File: los_keeper/CMakeLists.txt [hunk 1]
```diff
@@ -8,13 +8,13 @@ file(GLOB_RECURSE LIBRARY_HDRS "include/los_keeper/*.h")
...
-add_library(${PROJECT_NAME} STATIC ${LIBRARY_SRCS} ${LIBRARY_HDRS})
-target_link_libraries(${PROJECT_NAME} ${PCL_LIBRARIES})
+add_library(${PROJECT_NAME} STATIC ${LIBRARY_SRCS} ${LIBRARY_HDRS} )
+target_link_libraries(${PROJECT_NAME} ${PCL_LIBRARIES} pthread)
 target_include_directories(
   ${PROJECT_NAME}
   PUBLIC ${EIGEN_INCLUDE_DIR} ${PCL_INCLUDE_DIRS}
          $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/include>
-         $<INSTALL_INTERFACE:include>)
+         $<INSTALL_INTERFACE:include> )
```
[the remark this answers is not in the document]
Comment (YunwooLee94) [reply los_keeper/CMakeLists.txt:17]: Okay:)

**Comment by YunwooLee94** [comment]
@icsl-Jeon Yes, I rebased "utils_arrange" into "master". Thx.

**Comment by YunwooLee94** [comment]
@icsl-Jeon  Could you "approve" this PR?
I changed all the stuff you said.

**Comment by icsl-Jeon** [comment]
LGTM

**Pull request merged**